sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1409,7 +1425,12 @@ def _get_loss(self, sample_weight):
             )
             return _LOSSES["absolute_error"](sample_weight=sample_weight)
 
-        return _LOSSES[self.loss](sample_weight=sample_weight)
+        if self.loss == "quantile":
+            return _LOSSES[self.loss](
+                sample_weight=sample_weight, quantile=self.quantile
```

Comment:
Currently there is no error checking for `quantile=None`. Since `PinBall` loss already does some error checking:

https://github.com/scikit-learn/scikit-learn/blob/7861a4cb07f625f8828a95bd2a5ac524494b3f16/sklearn/_loss/loss.py#L607-L611

I think we can include the `None` check there.

Reply:
I added input checks and tests in https://github.com/scikit-learn/scikit-learn/pull/21800/commits/539c3a29839fce65a538e91a347bd0800b6e493c.
Could you have a short look over it?